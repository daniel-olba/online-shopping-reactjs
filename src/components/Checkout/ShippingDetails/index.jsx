import React from "react";
import { useCheckoutState } from "contexts/CheckoutContext";

// Components
import { Typography, IconButton, Button } from "@material-ui/core";
import { Check } from "@material-ui/icons";

const ShippingDetails = () => {
    const checkoutState = useCheckoutState();

    const { shippingData } = checkoutState.values;

    return (
        <>
            <Typography variant="h6" gutterBottom style={{ margin: "20px 0" }}>
                Shipping Details
            </Typography>
            <Typography variant="body2">
                {shippingData.firstName} {shippingData.lastName}
            </Typography>
            <Typography variant="body2">{shippingData.email}</Typography>
            <Typography variant="body2">{shippingData.phone}</Typography>
            <Typography variant="body2">
                {shippingData.address}, {shippingData.postal}
            </Typography>
            <Typography variant="body2">
                {shippingData.shippingCountry}
            </Typography>
            <br />
            {checkoutState.values.shippingDetailsSaved ? (
                <Typography variant="button">
                    <IconButton
                        color="inherit"
                        aria-label="menu"
                    >
                        <Check />
                    </IconButton>
                    Details Saved
                </Typography>
            ) : (
                <Button
                    variant="outlined"
                    onClick={() => checkoutState.saveShippingDetails()}
                >
                    Save shipping details
                </Button>
            )}

            <br />
        </>
    );
};

export default ShippingDetails;
