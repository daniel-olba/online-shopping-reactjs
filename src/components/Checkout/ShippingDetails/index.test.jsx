import React from "react";
import { mount } from "enzyme";

import renderer from "react-test-renderer";

import ContextsMock from "../../../common/all-contexts-mock";

import ShippingDetails from "./index";

// const checkoutMock = {
//     values: {
//         shippingData: {
//             firstName: "John",
//             lastName: "Doe",
//             email: "johndoe@gmail.com",
//             phone: "1234567890",
//             address: "Address test",
//             postal: "1234-567",
//             shippingCountry: "Country X",
//         },
//         shippingDetailsSaved: false,
//     },
//     actions: {},
//     saveShippingDetails: jest.fn().mockImplementation(),
// };

describe("ShippingDetails component unit testing", () => {
    function baseXML() {
        return ContextsMock({
            component: <ShippingDetails />,
            // checkoutContextValue: checkoutState,
        });
    }

    const tree = renderer.create(baseXML()).toJSON();

    it("matches snapshot", () => {
        expect(tree).toMatchSnapshot();
    });

    it("should render correctly", () => {
        expect(tree).toBeTruthy();
    });

    it("should render six Typography components and one button when shippingDetailsSaved is false", () => {
        const wrapper = mount(baseXML());
        // 1 H6 Typography variant
        expect(wrapper.find("h6")).toHaveLength(1);
        // 5 paragraph Typography variant
        expect(wrapper.find("p")).toHaveLength(5);
        // 1 button since shippingDetailsSaved is false
        expect(wrapper.find("button")).toHaveLength(1);
    });

    // it("should render seven Typography components and one icon when shippingDetailsSaved is true", () => {
    //     const wrapper = mount(
    //         baseXML({
    //             values: {
    //                 ...checkoutMock.values,
    //                 shippingDetailsSaved: true,
    //             },
    //         })
    //     );
    //     console.log('debug: ', wrapper.debug());
    //     // 1 H6 Typography variant
    //     expect(wrapper.find("h6")).toHaveLength(1);
    //     // 5 paragraph Typography variant
    //     expect(wrapper.find("p")).toHaveLength(5);
    //     // 1 button since shippingDetailsSaved is false
    //     // expect(wrapper.find('button')).toHaveLength(1);
    // });
});
