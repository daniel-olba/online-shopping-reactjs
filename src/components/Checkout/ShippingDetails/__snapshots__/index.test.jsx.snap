// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingDetails component unit testing matches snapshot 1`] = `
Array [
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
    style={
      Object {
        "margin": "20px 0",
      }
    }
  >
    Shipping Details
  </h6>,
  <p
    className="MuiTypography-root MuiTypography-body2"
  >
     
  </p>,
  <p
    className="MuiTypography-root MuiTypography-body2"
  />,
  <p
    className="MuiTypography-root MuiTypography-body2"
  />,
  <p
    className="MuiTypography-root MuiTypography-body2"
  >
    , 
  </p>,
  <p
    className="MuiTypography-root MuiTypography-body2"
  />,
  <br />,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Save shipping details
    </span>
  </button>,
  <br />,
]
`;
